OneCommunityGlobal/HighestGoodNetworkApp
# Shiwani Fix Log time entry for other users at other user's timelog page

# Description
<img width="786" alt="Screenshot 2023-10-12 at 3 52 24 PM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78906820/d754419e-cd55-45c4-baca-028786180f45">


## Related PRS (if any):

## Main changes explained:
- Created two separate user ids called LoggedInUserId and currUserId where in LoggedInUserId will have the user id of the logged in user and currUserId will have the user whose time log page is being visited. Added these two user ids in TimeEntry.jsx, Timer.jsx, TimeEntryForm.jsx.
- Added a new user profile variable called curruserProfile in TimeEntryForm.jsx which will have the user profile details of the logged in user which is present in redux store payload. This is required to avoid mixing up of details in mongoDB as initially userProfile was used but it contains the user profile details of the user whose time log page is being viewed which leads to mongo db update for the logged in user with the other user details.
- Added getUserProfile action with current user id instead of logged in user id, and getTimeEntriesForWeek action to update the redux store for the 0th week since redux store is updated with logged in user's 0th week details for TimeEntries state variable.
- Added curruserId to TimeEntry object under handleSubmit in TimeEntryForm.jsx
- Updated TimeEntries.js under actions for updateTimeEntriesForWeek action which in turn calls GetTimeEntriesForWeek by changing timeEntry.personId with timeEntry.curruserId so that the time entries are updated with user whose time log page is being viewed so that the frontend displays the time entries accordingly.

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. log in as any user
4. go to dashboard→Leaderboard->click the progress bar for any of the users listed under leaderboard
5. Record time in the chosen user's time log page
6. Verify that after recording time log, user profile details of logged in user does not change. Check if the time log details of logged in user does not display in other user's time log page.
7. Additionally, test by recording in own time log page by clicking on the progress bar adjacent to your name, and test by recording time in dashboard. The time log details will change for logged in user's time log page and dashboard in this case.
8. Feel free to leave any comments and screen recordings of your test.

## Screenshots or videos of changes:

Before: 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78906820/ed1b6584-f9a1-4437-93c8-fe269d4c0fb4

After: 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78906820/5e9b0c38-2707-47b1-b006-0860077949ae



## Note:
- Time log will get updated for logged in user's time log page and when user records time log in dashboard.
- Time log details of logged in user should not appear if you are viewing someone else's time log page.
